### ISSUE TITLE
Report contents review

### ISSUE THREAD

**Issue by tingyuw** [JacquesCarette/Drasil#3200]
First draft of the table of contents and some thoughts of what will be covered in each chapter. 
### Chapter 1 Introduction
- Drasil overview and introduction.
- Why do we want to generate Jupyter notebook in Drasil? 
- What do we want to achieve?
### Chapter 2 Generate a basic notebook example in Drasil
- Jupyter notebook overview
- Notebook structure. (Building a general structure)
- generate projectile motion lesson example
- generate SRS in .ipynb format
### Chapter 3 Problems of the current Drasil (problems I faced when trying to generate the notebook)
- Cells. No cell concept in current Drasil. What is a "cell" and how do we want to define it? It should not be too large to maintain yet too small to cover insufficient information.
- Nested vs. linear/flatten. (focus of this chapter) Pros and cons of each. Comparison of two structures. Why this topic exists and why try to flatten the structure? Several approaches has brought up and what is the final implementation? 
- Math generation (can be extend to what is insufficient of the existing printer)
### Chapter 4 Code generation
Haven't start it but still plan on completing.
### Chapter 5 Future work
- More concrete notebook structure.
- etc.

**Comment by smiths** [comment]
@tingyuw, if your intention is to have your table of contents reviewed, you should assign a reviewer, through the assignee option in GitHub.  Since this is an issue, and the only way to close it is to provide a review, I'm going to assume that your intention was to assign either myself or @JacquesCarette (or both) as reviewers.  I'll provide a review below and then close the issue.  @JacquesCarette or you can reopen the issue if there are further questions or comments.

**Chapter 1 Introduction**

I like the topics you have listed, but I suggest you change the order in which you present the material.  I also have a suggestion for further content.

For the change in order, you make your report about Drasil first.  I think you will find the report is easier to write if you make it about the problem first.  I think you want to start with your problem solution, something like: 

"Jupyter notebooks can be generated from codified knowledge to improve their reusability, usability and modifiability."

This is what you want to achieve.  How you achieve it (using Drasil) is part of the how.  The role of the problem solution in your report is similar to the role of the [thesis statement in a thesis proposal](http://jeapostrophe.github.io/2013-03-26-thesis-p-post.html).

Your introduction should also include background information on Jupyter notebooks and generative programming.  You will also want to motivate how a generative approach improves on the current approach of manually writing notebooks.

**Chapters 2 and 3**

I like your idea of starting out with a basic notebook in Chapter 2 and then showing what you need to add/modify in Drasil to improve the notebook in Chapter 3.  You will want to be very clear on the scope of the types of problems you can do using the basic approach.  The limitations of this approach will then be your motivation for Chapter 3.  

Your title for Chapter 3 suggests you are presenting problems with the current Drasil, but you are also presenting solutions to these problems, right?

**Chapter 4**

By code generation, I'm assuming that you mean adding the ability to generate notebooks that include both text and code?  We really do want to be able to mix text and code, or the Jupyter notebook example will feel very incomplete.

**Chapter 5 and 6**

You might want to include the future work as part of the Conclusions chapter.

You can find additional information on writing, including the [document by @JacquesCarette](https://gitlab.cas.mcmaster.ca/SEforSC/se4sc/-/blob/git-svn/GradStudResources/WritingGuidelines/WritingProposals_JC.docx) that points to the thesis proposal link used above, in the [se4sc repo](https://gitlab.cas.mcmaster.ca/SEforSC/se4sc/-/tree/git-svn/GradStudResources/WritingGuidelines).

**Comment by tingyuw** [comment]
Sorry, I thought I have assigned reviewers.

Thank you @smiths for the comments and the additional resources. To answer your questions and for clarifications:
- Yes for chapter 3, the final solution and implementation will be presented and even some approaches that we discussed or tried to implement but failed will also be presented. Since I was thinking the process is part of the research, I've tried a few implementations back and forth on some issues and would like to include those as well.
- Yes, by code generation, I mean the ability to add code in the notebook. The projectile lesson example is the prototype notebook artifact, the structure I built for notebook is based on this so we can circle back to this example in chapter 4 after we can generate notebook with a mix of text and code.
